homeassistant/components/obihai/sensor.py

```diff
@@ -46,22 +46,34 @@ def setup_platform(hass, config, add_entities, discovery_info=None):
 
     pyobihai = PyObihai()
 
-    services = pyobihai.get_state(host, username, password)
+    login = pyobihai.check_account(host, username, password)
+    if login:
```

Comment:
Invert this check and log error and `return` if true. Then we can outdent below.

Reply:
ok